samza-sql/src/main/java/org/apache/samza/sql/avro/ConfigBasedAvroRelConverterFactory.java

```diff
@@ -0,0 +1,59 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.apache.samza.sql.avro;
+
+import java.util.Optional;
+import java.util.Properties;
+
+import org.apache.samza.sql.interfaces.SamzaRelConverter;
+import org.apache.samza.sql.interfaces.SamzaRelConverterFactory;
+
+
+/**
+ * Avro Schema Resolver that uses static config to return a schema for a SystemStream.
+ * Schemas are configured using the config of format {systemName}.{streamName}.schema.
+ */
+public class ConfigBasedAvroRelConverterFactory implements SamzaRelConverterFactory {
+
+  public static final String CFG_SOURCE_SCHEMA = "%s.%s.schema";
```

Comment:
As we discussed, it is not possible for Samza Sql to follow the stream convention that got added to the samza recently because Samza SQL as a service needs to support the sql statements at runtime, it might not know all the streams that it needs to support when the service is coming up.